pkg/apis/resource/validation/validation.go

```diff
@@ -17,172 +17,260 @@ limitations under the License.
-func ValidateClaimUpdate(resourceClaim, oldClaim *resource.ResourceClaim) field.ErrorList {
-	allErrs := corevalidation.ValidateObjectMetaUpdate(&resourceClaim.ObjectMeta, &oldClaim.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(resourceClaim.Spec, oldClaim.Spec, field.NewPath("spec"))...)
-	allErrs = append(allErrs, ValidateClaim(resourceClaim)...)
+func validateDeviceConfiguration(config resource.DeviceConfiguration, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if config.Opaque == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("opaque"), ""))
+	} else {
+		allErrs = append(allErrs, validateOpaqueConfiguration(*config.Opaque, fldPath.Child("opaque"))...)
+	}
 	return allErrs
 }
 
-// ValidateClaimStatusUpdate tests if an update to the status of a ResourceClaim is valid.
-func ValidateClaimStatusUpdate(resourceClaim, oldClaim *resource.ResourceClaim) field.ErrorList {
-	allErrs := corevalidation.ValidateObjectMetaUpdate(&resourceClaim.ObjectMeta, &oldClaim.ObjectMeta, field.NewPath("metadata"))
-	fldPath := field.NewPath("status")
-	// The name might not be set yet.
-	if resourceClaim.Status.DriverName != "" {
-		allErrs = append(allErrs, validateResourceDriverName(resourceClaim.Status.DriverName, fldPath.Child("driverName"))...)
-	} else if resourceClaim.Status.Allocation != nil {
-		allErrs = append(allErrs, field.Required(fldPath.Child("driverName"), "must be specified when `allocation` is set"))
+func validateOpaqueConfiguration(config resource.OpaqueDeviceConfiguration, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateDriverName(config.Driver, fldPath.Child("driver"))...)
+	// Validation of RawExtension as in https://github.com/kubernetes/kubernetes/pull/125549/
+	var v any
+	if len(config.Parameters.Raw) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("parameters"), ""))
+	} else if err := json.Unmarshal(config.Parameters.Raw, &v); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("parameters"), "<value omitted>", fmt.Sprintf("error parsing data: %v", err.Error())))
+	} else if v == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("parameters"), ""))
+	} else if _, isObject := v.(map[string]any); !isObject {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("parameters"), "<value omitted>", "parameters must be a valid JSON object"))
 	}
 
-	allErrs = append(allErrs, validateAllocationResult(resourceClaim.Status.Allocation, fldPath.Child("allocation"))...)
-	allErrs = append(allErrs, validateResourceClaimConsumers(resourceClaim.Status.ReservedFor, resource.ResourceClaimReservedForMaxSize, fldPath.Child("reservedFor"))...)
+	return allErrs
+}
+
+func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaimStatus, claimDeleted bool, requestNames sets.Set[string], fldPath *field.Path) field.ErrorList {
```

Comment:
Does this need the word "update" in the func name?

Reply:
Doesn't need to be, but seems to more common:
```console
$ git grep -w func.*Validate.*StatusUpdate pkg/apis | grep -v _test.go | wc -l
24
$ git grep -w func.*Validate.*Status pkg/apis | grep -v _test.go | wc -l
12
```